python/dgl/data/knowledge_graph.py

```diff
@@ -0,0 +1,737 @@
+from __future__ import absolute_import
+
+import numpy as np
+import pickle as pkl
+import networkx as nx
+import scipy.sparse as sp
+import os, sys
+
+from .dgl_dataset import DGLBuiltinDataset
+from .utils import download, extract_archive, get_download_dir
+from .utils import save_graphs, load_graphs, save_info, load_info, makedirs, _get_dgl_url
+from .utils import generate_mask_tensor
+from .utils import deprecate_property, deprecate_function
+from ..utils import retry_method_with_fix
+from .. import backend as F
+from ..graph import DGLGraph
+from ..graph import batch as graph_batch
+from ..convert import graph as dgl_graph
+
+class KnowledgeGraphDataset(DGLBuiltinDataset):
+    """KnowledgeGraph link prediction dataset
+
+    The dataset contains a graph depicting the connectivity of a knowledge
+    base. Currently, the knowledge bases from the
+    `RGCN paper <https://arxiv.org/pdf/1703.06103.pdf>`_ supported are
+    FB15k-237, FB15k, wn18
+
+    Parameters
+    -----------
+    name: str
+        Name can be 'FB15k-237', 'FB15k' or 'wn18'.
+    reverse: boo
+        Whether add reverse edges. Default: True.
+    raw_dir : str
+        Raw file directory to download/contains the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+      Whether to print out progress information. Default: True.
+    """
+    def __init__(self, name, reverse=True, raw_dir=None, force_reload=False, verbose=True):
+        self._name = name
+        self.reverse = reverse
+        url = _get_dgl_url('dataset/') + '{}.tgz'.format(name)
+        super(KnowledgeGraphDataset, self).__init__(name,
+                                                    url=url,
+                                                    raw_dir=raw_dir,
+                                                    force_reload=force_reload,
+                                                    verbose=verbose)
+
+    def download(self):
+        r""" Automatically download data and extract it.
+        """
+        tgz_path = os.path.join(self.raw_dir, self.name + '.tgz')
+        download(self.url, path=tgz_path)
+        extract_archive(tgz_path, self.raw_path)
+
+    def process(self):
+        """
```

Comment:
I'm a bit confused. what is the purpose of the list of params?

Reply:
I think I need to change the description.